From 7b12d7d9b86964654b81b64981d991f0301d8434 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Tue, 19 Aug 2025 12:54:28 +0200
Subject: [PATCH] Ensure that the CMake build type is respected alsof for
 multi-config generatros

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 895b3798..24f480c0 100644
--- a/setup.py
+++ b/setup.py
@@ -135,9 +135,9 @@ def _build_all_extensions_with_cmake(self):
             ["cmake", str(_ROOT_DIR)] + cmake_args, cwd=self.build_temp
         )
         print("Calling cmake --build", flush=True)
-        subprocess.check_call(["cmake", "--build", "."], cwd=self.build_temp)
+        subprocess.check_call(["cmake", "--build", ".","--config",cmake_build_type], cwd=self.build_temp)
         print("Calling cmake --install", flush=True)
-        subprocess.check_call(["cmake", "--install", "."], cwd=self.build_temp)
+        subprocess.check_call(["cmake", "--install", ".","--config",cmake_build_type], cwd=self.build_temp)
 
     def copy_extensions_to_source(self):
         """Copy built extensions from temporary folder back into source tree.
