diff --git a/setup.py b/setup.py
index 59b5ef53..cb571fbb 100644
--- a/setup.py
+++ b/setup.py
@@ -123,6 +123,7 @@ def _build_all_extensions_with_cmake(self):
             f"-DCMAKE_BUILD_TYPE={cmake_build_type}",
             f"-DPYTHON_VERSION={python_version.major}.{python_version.minor}",
             f"-DENABLE_CUDA={enable_cuda}",
+            "-DPYBIND_OPS_MODULE_NAME=core_pybind_ops"
             f"-DTORCHCODEC_DISABLE_COMPILE_WARNING_AS_ERROR={torchcodec_disable_compile_warning_as_error}",
         ]
 
