From b75ecfb919f0c0820b5b2292fd9d716f7a6bde68 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sun, 1 Feb 2026 11:56:49 +0100
Subject: [PATCH] Skip vpx tests in ffmpeg 8

---
 test/test_encoders.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/test_encoders.py b/test/test_encoders.py
index fef19ac99..e3cae6d2d 100644
--- a/test/test_encoders.py
+++ b/test/test_encoders.py
@@ -937,7 +937,7 @@ def encode_to_tensor(frames):
                    pytest.mark.slow,
                    pytest.mark.skipif(
                        get_ffmpeg_major_version() == 4
-                        or (IS_WINDOWS and get_ffmpeg_major_version() in (6, 7)),
+                        or (IS_WINDOWS and get_ffmpeg_major_version() >= 6),
                        reason="Codec for webm is not available in this FFmpeg installation.",
                    ),
                ],
@@ -992,7 +992,7 @@ def test_round_trip(self, tmp_path, format, method):
                    pytest.mark.slow,
                    pytest.mark.skipif(
                        get_ffmpeg_major_version() == 4
-                        or (IS_WINDOWS and get_ffmpeg_major_version() in (6, 7)),
+                        or (IS_WINDOWS and get_ffmpeg_major_version() >= 6),
                        reason="Codec for webm is not available in this FFmpeg installation.",
                    ),
                ],
@@ -1037,7 +1037,7 @@ def test_against_to_file(self, tmp_path, format, method):
                    pytest.mark.slow,
                    pytest.mark.skipif(
                        get_ffmpeg_major_version() == 4
-                        or (IS_WINDOWS and get_ffmpeg_major_version() in (6, 7)),
+                        or (IS_WINDOWS and get_ffmpeg_major_version() >= 6),
                        reason="Codec for webm is not available in this FFmpeg installation.",
                    ),
                ],
